File: functions/ffmpeg/variants/go/3_weight_8.go
package main

func weight(i int32, blen int32, offset int32) int32 {
    switch {
    case i < 2*offset:
        switch offset {
        case 1:
            switch i {
            case 0:
                return 3
            default:
                return 5
            }
        default:
            return 1 + (6*i + offset - 1)/(2*offset - 1)
        }
    case i > blen-1-2*offset:
        switch offset {
        case 1:
            switch blen-1-i {
            case 0:
                return 3
            default:
                return 5
            }
        default:
            return 1 + (6*(blen-1-i) + offset - 1)/(2*offset - 1)
        }
    default:
        return 8
    }
}


func main(){
    weight(0,0,0)
}

File: functions/ffmpeg/variants/c/3_weight_8.c
#include <stdint.h>

static int weight(int i, int blen, int offset)
{
    if (i < 2 * offset) {
        return (offset == 1) ? ((i) ? 5 : 3) : (1 + (6 * i + offset - 1) / (2 * offset - 1));
    } else if (i > blen - 1 - 2 * offset) {
        return (offset == 1) ? ((blen - 1 - i) ? 5 : 3) : (1 + (6 * (blen - 1 - i) + offset - 1) / (2 * offset - 1));
    }
    return 8;
}


int main(){
    weight(0,0,0);
}